is_spinor_ is a private member in an assert statement

**Describe the bug**

```
/builds/correaa/boost-mpi3/qmcpack/src/QMCWaveFunctions/Fermion/MultiDiracDeterminant.cpp: In member function 'virtual qmcplusplus::WaveFunctionComponent::LogValueType qmcplusplus::MultiDiracDeterminant::updateBuffer(qmcplusplus::ParticleSet&, qmcplusplus::WaveFunctionComponent::WFBufferType&, bool)':
/builds/correaa/boost-mpi3/qmcpack/src/QMCWaveFunctions/Fermion/MultiDiracDeterminant.cpp:270:12: error: 'bool qmcplusplus::ParticleSet::is_spinor_' is protected within this context
  270 |   assert(P.is_spinor_ == is_spinor_);
```

**To Reproduce**

compile with DEBUG flags.

Is the github CI compiling in debug mode at all?

**Expected behavior**

should compile. Be public for example.

**System:**

Debian testing

**Additional context**

https://gitlab.com/correaa/boost-mpi3/-/jobs/1924471860#L1734

--- a/src/QMCWaveFunctions/Fermion/MultiDiracDeterminant.2.cpp
+++ b/src/QMCWaveFunctions/Fermion/MultiDiracDeterminant.2.cpp
@@ -343,7 +343,7 @@ void MultiDiracDeterminant::evaluateDetsAndGradsForPtclMove(const ParticleSet& P
 
 void MultiDiracDeterminant::evaluateDetsAndGradsForPtclMoveWithSpin(const ParticleSet& P, int iat)
 {
-  assert(P.is_spinor_ == is_spinor_);
+  assert(P.isSpinor() == is_spinor_);
   UpdateMode = ORB_PBYP_PARTIAL;
   evalOrb1Timer.start();
   Phi->evaluateVGL_spin(P, iat, psiV, dpsiV, d2psiV, dspin_psiV);
@@ -501,7 +501,7 @@ void MultiDiracDeterminant::evaluateGrads(ParticleSet& P, int iat)
 
 void MultiDiracDeterminant::evaluateGradsWithSpin(ParticleSet& P, int iat)
 {
-  assert(P.is_spinor_ == is_spinor_);
+  assert(P.isSpinor() == is_spinor_);
   const int WorkingIndex = iat - FirstIndex;
   assert(WorkingIndex >= 0 && WorkingIndex < LastIndex - FirstIndex);
 

--- a/src/QMCWaveFunctions/Fermion/MultiDiracDeterminant.cpp
+++ b/src/QMCWaveFunctions/Fermion/MultiDiracDeterminant.cpp
@@ -267,7 +267,7 @@ MultiDiracDeterminant::LogValueType MultiDiracDeterminant::updateBuffer(Particle
                                                                         WFBufferType& buf,
                                                                         bool fromscratch)
 {
-  assert(P.is_spinor_ == is_spinor_);
+  assert(P.isSpinor() == is_spinor_);
   if (is_spinor_)
     evaluateForWalkerMoveWithSpin(P, (fromscratch || UpdateMode == ORB_PBYP_RATIO));
   else
@@ -290,7 +290,7 @@ MultiDiracDeterminant::LogValueType MultiDiracDeterminant::updateBuffer(Particle
 
 void MultiDiracDeterminant::copyFromBuffer(ParticleSet& P, WFBufferType& buf)
 {
-  assert(P.is_spinor_ == is_spinor_);
+  assert(P.isSpinor() == is_spinor_);
   buf.get(psiM.first_address(), psiM.last_address());
   buf.get(FirstAddressOfdpsiM, LastAddressOfdpsiM);
   buf.get(d2psiM.first_address(), d2psiM.last_address());
@@ -319,7 +319,7 @@ void MultiDiracDeterminant::acceptMove(ParticleSet& P, int iat, bool safe_to_del
 {
   const int WorkingIndex = iat - FirstIndex;
   assert(WorkingIndex >= 0 && WorkingIndex < LastIndex - FirstIndex);
-  assert(P.is_spinor_ == is_spinor_);
+  assert(P.isSpinor() == is_spinor_);
   log_value_ref_det_ += convertValueToLog(curRatio);
   curRatio = ValueType(1);
   switch (UpdateMode)
@@ -469,7 +469,7 @@ MultiDiracDeterminant::~MultiDiracDeterminant() = default;
 
 void MultiDiracDeterminant::registerData(ParticleSet& P, WFBufferType& buf)
 {
-  assert(P.is_spinor_ == is_spinor_);
+  assert(P.isSpinor() == is_spinor_);
 
   //extra pointers
   FirstAddressOfGrads = &(grads(0, 0)[0]);

